src/Sentry/Internal/Http/CachingTransport.cs

```diff
@@ -80,6 +82,20 @@ private void Initialize(bool startWorker)
 
             // Start a worker, if one is needed
             _worker = startWorker ? Task.Run(CachedTransportBackgroundTaskAsync) : Task.CompletedTask;
+
+            // Wait for init timeout, if configured.  (Can't do this without a worker.)
+            if (startWorker && _options.InitCacheFlushTimeout > TimeSpan.Zero)
+            {
+                using (_initCacheResetEvent = new ManualResetEventSlim())
+                {
+                    // This will complete either when the first round of processing is done,
+                    // or on timeout, whichever comes first.
+                    _initCacheResetEvent.Wait(_options.InitCacheFlushTimeout);
```

Comment:
Is it worth adding this within a `if` and logging the outcome?

Reply:
Probably.  Will do. Thanks.